### ISSUE TITLE
UEFI DXE driver status: Bridge Found (0x000000000000003C)

### ISSUE THREAD

**Issue by Uranium2010** [terminatorul/NvStrapsReBar#79]
**System**
 - Motherboard: Asus X571GT
 - BIOS Version: 312
 - GPU: GTX 1650 Mobile 10DE:1F91 1043:1A81
 - [x] CSM is turned off. **Make sure to confirm this in the BIOS and not with GPU-Z or similar since it can be inaccurate**
 - [x] 4G decoding is enabled. **Make sure to confirm this in the BIOS and not with GPU-Z or similar since it can be inaccurate**
 - [ ] UEFIPatch is applied (see [Using UEFIPatch](https://github.com/xCuri0/ReBarUEFI/wiki/Using-UEFIPatch) for more information). On some motherboards [DSDT Patching](https://github.com/xCuri0/ReBarUEFI/wiki/DSDT-Patching) is also needed
 - [x] I have read [Common issues (and fixes)](https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes))

**Description**

Hello,
I added the module to the BIOS, I tried to patch but apparently no need.
I activated "4GB MMIO BIOS assignment" using modGRUBShell.efi, the CMS is deactivated.
When testing different combinations in NvStrapsReBar, "Current BAR Size" remains at 256 and UEFI DXE driver status: Bridge Found (0x000000000000003C).

![Sans titre](https://github.com/terminatorul/NvStrapsReBar/assets/68108402/d30f8191-2e1d-49c6-b133-e6e20a0495a8)

**Comment by pexcfequinnet** [comment]
Have you tried setting the BAR size using option (P)?

**Comment by Uranium2010** [comment]
> Have you tried setting the BAR size using option (P)?

Yes, I tested all kinds of combinations, I always have the same status for the driver DXE.

The status is as it should be when the driver is not configured or there is a change in the BIOS, but as soon as I configure it and restart the computer, I have the status "Bridge Found".

I looked, we have the same chipset but BIOS are different. I will test old versions.

**Comment by terminatorul** [comment]
"Bridge Found" means the GPU is NOT found by the DXE driver, despite the Windows executable can find in Device Manager... :(

This looks like an issue in the latest release v0.4. Two other users reported a similar issue  https://github.com/terminatorul/NvStrapsReBar/issues/77, and I do not know where the problem is.

Can you try again with v0.3 ?

With v0.3, you have to remember the DXE driver depends on the settings in UEFI setup, and should be disabled before changing UEFI settings, and re-enabled after.

**Comment by Uranium2010** [comment]
> Can you try again with v0.3 ?
Same with v0.3 and v0.4-rc1.

**Comment by Uranium2010** [comment]
With BIOS 305
![bar305](https://github.com/terminatorul/NvStrapsReBar/assets/68108402/364140c4-b1e1-467d-83cd-10027ba2e497)

**Comment by adrianconstantin-leroy** [comment]
Sorry I cannot tell what the issue is.

It looks like the GPU device is not available during early boot (or does not have the PCI IDs configured), and is only enabled by motherboard firmware or vBIOS later in the boot process. This is just a guess 'though, from the behavior I can see. I do not actually know what the firmware for that laptop does... might also be a small hardware problem with the GPU board, that the driver knows how to work around

**Comment by tribhoon** [comment]
Hi, I am sorry for reviving this old issue but I have a similar Asus model and have the same problem, the DXE driver stays at "Bridge Found". I think you are right, the problem is that the GPU isn't woken up by the firmware early in the boot process not allowing the driver to configure the BAR size. Were you able to get it working @Uranium2010 ?

Any help would be really appreciated, thank you.

<img width="1918" height="1078" alt="Image" src="https://github.com/user-attachments/assets/d8f7b409-fe4f-4153-adcb-b2bdac64bb16" />

**Comment by Uranium2010** [comment]
Hello,
Sorry for the late reply @tribhoon , unfortunately I can't help you any further, I've tried a lot of things without finding a solution.
By doing everything wrong, I softbricked the laptop (incorrect setting in NVRAM), it's now unusable, it didn't appreciate the CH431A clip.

**Comment by tribhoon** [comment]
Damn, I see, sorry to hear that.
